easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -497,25 +502,23 @@ def include_defined_parameters(keyset):
             for group in keyset:
                 printed = False
                 for key in group:
-                    val = self._config[key][0]
-                    def_val = DEFAULT_CONFIG[key]
-                    if val not in def_val:
-                        ebtxt.append("%s = %s" % (key, quote_str(val, escape_newline=True)))
+                    if self[key] != default_values[key]:
+                        ebtxt.append("%s = %s" % (key, quote_str(self[key], escape_newline=True)))
                         printed_keys.append(key)
                         printed = True
                 if printed:
                     ebtxt.append("")
 
-
         # print easyconfig parameters ordered and in groups specified above
         ebtxt = []
         printed_keys = []
         include_defined_parameters(grouped_keys)
 
         # print other easyconfig parameters at the end
-        for key, [val, _, _] in sorted(DEFAULT_CONFIG.items()):
-            if key not in printed_keys and key not in last_keys and val != self._config[key][0]:
-                ebtxt.append("%s = %s" % (key, quote_str(self._config[key][0], escape_newline=True)))
+        keys_to_ignore = printed_keys + last_keys
+        for key in default_values:
+            if key not in keys_to_ignore and default_values[key] != self[key]:
```

Comment:
replace `default_values[key] != self[key]` with `self[key] != default_values[key]`, so that it matches the check above, and it makes more sense too: "is value equal to the default value"